packages/@blockprotocol/graph/src/custom-element.ts

```diff
@@ -57,28 +53,34 @@ export abstract class BlockElementBase<
    * A helper method to update the properties of the entity loaded into the block, i.e. this.graph.blockEntity
    * @param properties the properties object to assign to the entity, which will overwrite the existing object
    */
-  protected updateSelf(properties: NonNullable<BlockEntityProperties>) {
+  protected updateSelf(properties: EntityPropertiesObject) {
     if (!this.graphService) {
       throw new Error("Cannot updateSelf – graphService not yet connected.");
     }
     if (!this.graph) {
       throw new Error(
         "Cannot update self: no 'graph' property object passed to block.",
       );
-    } else if (!this.graph.blockEntity) {
+    } else if (!this.graph.blockEntitySubgraph) {
       throw new Error(
-        "Cannot update self: no 'blockEntity' on 'graph' object passed to block",
+        "Cannot update self: no 'blockEntitySubgraph' on 'graph' object passed to block",
       );
-    } else if (!this.graph.blockEntity.metadata) {
+    }
+
+    const blockEntity = getRoots(this.graph.blockEntitySubgraph)[0];
+    if (!blockEntity) {
       throw new Error(
-        "Cannot update self: no 'metadata' on graph.blockEntity passed to block",
+        "Cannot update self: no root entity on graph.blockEntitySubgraph passed to block",
       );
     }
 
     return this.graphService.updateEntity({
       data: {
-        entityId: this.graph.blockEntity.metadata.editionId.baseId,
+        entityId: blockEntity.metadata.editionId.baseId,
+        entityTypeId: blockEntity.metadata.entityTypeId,
         properties,
+        leftToRightOrder: blockEntity.linkData?.leftToRightOrder,
+        rightToLeftOrder: blockEntity.linkData?.rightToLeftOrder,
```

Comment:
As it stands it's impossible to update the link order via this method if the root entity is a link entity. We could add in additional arguments for order to this

We could also consider removing this method as it's just a helper to make calling `updateEntity` for the root entity a bit easier

Reply:
No, it's fine, we can clean up later and will need to revisit this in light of whatever we do with generics in `Subgraph` anyway

Reply:
😅  I jumped the gun and did it beforehand, I'm happy to drop the commit but I think it's better to have it be consistent in the interim anyway

Reply:
No worries, sounds good!